Platform defaults to host, even if it is not expected

If to forget set `CONFIG_PLATFORM` variable in CMake project file then build fails due to incorrect linkage and lack of `platform` library. 

Steps to reproduce:
- create embedded project based on theCore
- do not set `CONFIG_PLATFORM`
- try to build the project

Expected:
- Error reported that emphasizes lack of platform name.

Actual: 
Note that t
- Vague error occurs, something like this:

```
/usr/lib/gcc/arm-none-eabi/6.2.0/../../../../arm-none-eabi/lib/libc.a(lib_a-exit.o): In function `exit':
exit.c:(.text.exit+0x2c): undefined reference to `_exit'
```

Issue rises because of default platform policy. If `CMAKE_PLATFORM` is not set then `host` platform is chosen. The policy must take cross-compilation into the account when choosing default platform.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -17,19 +17,29 @@ include(Findcppcheck)
 include(CppcheckTargets)
 
 # TODO: add same configuration mechanism for choosing right kernel
+
 message(STATUS "Checking [CONFIG_PLATFORM]...")
-if(NOT DEFINED CONFIG_PLATFORM)
-    message(STATUS "CONFIG_PLATFORM is not set, defaulting to 'host'...")
-    set(PLATFORM_NAME host)
+
+if(NOT CMAKE_CROSSCOMPILING)
+    if(DEFINED CONFIG_PLATFORM)
+        if (NOT CONFIG_PLATFORM STREQUAL "host")
+            message(FATAL_ERROR "CONFIG_PLATFORM must be set to 'host' if not cross-compiling")
+        endif()
+    else()
+        # It is fine to omit platform name if not cross-compiling
+        message(STATUS "CONFIG_PLATFORM not set, defauluting to 'host'...")
+        set(CONFIG_PLATFORM "host")
+    endif()
 else()
-    message(STATUS "Platform will be used: ${CONFIG_PLATFORM}")
-    set(PLATFORM_NAME ${CONFIG_PLATFORM}) # For convinience
+    # When cross-compiling, platform must be set explicitly.
+    if(NOT DEFINED CONFIG_PLATFORM)
+        message(FATAL_ERROR "CONFIG_PLATFORM must be set when cross-compiling")
+    endif()
 endif()
 
-set(CMAKE_C_FLAGS
-        "${CMAKE_C_FLAGS} ${CC_WARN_FLAGS} ${CC_EXTRA_FLAGS}")
-set(CMAKE_CXX_FLAGS
-        "${CMAKE_CXX_FLAGS} ${CC_WARN_FLAGS} ${CXX_EXTRA_FLAGS}")
+message(STATUS "Platform will be used: ${CONFIG_PLATFORM}")
+
+set(PLATFORM_NAME ${CONFIG_PLATFORM}) # For convinience
 
 # Linker definitions is propagated by the platform.
 include(${CORE_DIR}/platform/common/linker.cmake)

